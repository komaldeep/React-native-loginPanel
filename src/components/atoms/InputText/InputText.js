import React, {Component} from 'react';
import { StyleSheet, TextInput } from 'react-native';
import multipleStyles from 'react-native-multiple-styles'
const Dimensions = require('Dimensions');
import PropTypes from 'prop-types';

const viewport = {
  fullWidth: Dimensions.get('window').width,
  fullHeight: Dimensions.get('window').height,
};

export default class InputText extends Component {

  constructor(props) {
    super(props);
    this.state = {
      text: ''
    };
  }

  static propTypes = {
    placeholder: PropTypes.string,
    secureTextEntry: PropTypes.bool,
    inputFieldValue: PropTypes.func,
  };


  static defaultProps = {
    secureTextEntry: false,
  }

  inputValue = (text)=> {
    this.setState({
      text
    })
   this.props.inputFieldValue(text);
  }

  render() {
    const {
      placeholder,
      secureTextEntry
    } = this.props;
    return (
      <TextInput
        secureTextEntry={secureTextEntry}
        underlineColorAndroid="transparent"
        value={this.state.text}
        style={styles.textInput}
        onChangeText={this.inputValue}
        placeholder={placeholder}
      />
    );
  }
}


const styles = StyleSheet.create({
  textInput:{
    height: 40,
    borderBottomColor: 'black',
    borderBottomWidth: 2,
    marginBottom: 10,
    width: viewport.fullWidth* 0.80,
  }
});