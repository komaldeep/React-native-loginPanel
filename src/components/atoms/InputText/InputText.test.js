import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import InputText from './InputText';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const rendered = renderer.create(<InputText placeholder="placeholder for tests"/>).toJSON();
  expect(rendered).toMatchSnapshot();
});

// test('on change of Type input', () => {
//   const wrapper = mount(
//     <InputText
//       placeholder="placeholder for tests"
//       inputFieldValue={()=>{}}
//     />
//   );
//   wrapper.find('TextInput').simulate('onChangeText', { target: {
//     text: 'Change function' },
//   });
//   expect(wrapper.state().text).toEqual('Change function');
// });
