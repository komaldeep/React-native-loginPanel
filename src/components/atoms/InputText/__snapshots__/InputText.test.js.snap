// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="placeholder for tests"
  secureTextEntry={false}
  style={
    Object {
      "borderBottomColor": "black",
      "borderBottomWidth": 2,
      "height": 40,
      "marginBottom": 10,
      "width": 600,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
